File: frontend/analyze_eslint.js
const fs = require('fs');
try {
    const data = JSON.parse(fs.readFileSync('eslint_report.json', 'utf8'));
    const counts = {};
    let e = 0, w = 0;
    data.forEach(f => {
        (f.messages || []).forEach(m => {
            const id = m.ruleId || 'unknown_rule';
            counts[id] = (counts[id] || 0) + 1;
            if (m.severity === 2) e++; else w++;
        });
    });
    console.log('Total Errors: ' + e);
    console.log('Total Warnings: ' + w);
    console.log('Total IDE Issues: ' + (e + w));
    console.log('\nBreakdown by Rule:');
    const sorted = Object.entries(counts).sort((a,b) => b[1] - a[1]);
    sorted.forEach(([k,v]) => console.log(' - ' + k + ': ' + v));
} catch (err) {
    console.error(err);
}

File: analyze_eslint.py
import subprocess
import json
from collections import Counter

def run():
    try:
        # Run eslint with JSON formatter so we can easily parse the results
        result = subprocess.run(
            ["npx", "eslint", "src", "--ext", ".ts,.tsx", "--format", "json"], 
            cwd="frontend", 
            capture_output=True, 
            text=True
        )
        
        # ESLint exits with code 1 if there are errors, so we don't check returncode for success
        try:
            data = json.loads(result.stdout)
        except json.JSONDecodeError:
            print("Failed to parse ESLint output. Raw stdout:")
            print(result.stdout[:1000])
            return
            
        rule_counts = Counter()
        total_errors = 0
        total_warnings = 0
        
        for file in data:
            for message in file.get('messages', []):
                rule_id = message.get('ruleId') or 'parse_error'
                rule_counts[rule_id] += 1
                if message.get('severity') == 2:
                    total_errors += 1
                else:
                    total_warnings += 1
                    
        print(f"Total Errors: {total_errors}")
        print(f"Total Warnings: {total_warnings}")
        print(f"Total Pending IDE Issues: {total_errors + total_warnings}\n")
        print("Breakdown by ESLint Rule:")
        for rule, count in rule_counts.most_common():
            print(f" - {rule}: {count}")
            
    except Exception as e:
        print(f"Error executing ESLint analysis: {e}")

if __name__ == "__main__":
    run()
